src/cli/mod.rs

```diff
@@ -228,7 +361,28 @@ pub fn parse_and_run() -> Result<()> {
         match Cli::parse().command {
             Command::Repl(repl_args) => repl_args.into_cli().run(),
             Command::Load(load_args) => load_args.into_cli().run(),
-            // Command::Verify(verify_args) => verify_proof(&verify_args.proof_file),
+            #[allow(unused_variables)]
+            Command::Verify(verify_args) => {
+                #[cfg(not(target_arch = "wasm32"))]
+                {
+                    use pasta_curves::vesta;
+                    match verify_args.field {
+                        None => verify_proof::<pallas::Scalar>(&verify_args.proof_id),
+                        Some(field_str) => match parse_field(&field_str)? {
+                            LanguageField::Pallas => {
+                                verify_proof::<pallas::Scalar>(&verify_args.proof_id)
+                            }
+                            LanguageField::Vesta => {
+                                verify_proof::<vesta::Scalar>(&verify_args.proof_id)
+                            }
+                            LanguageField::BLS12_381 => {
+                                verify_proof::<blstrs::Scalar>(&verify_args.proof_id)
+                            }
+                        },
+                    }?;
```

Comment:
Are we determining everything from the field string here? What about bn254, which could work for SnarkPack and Nova?

Reply:
This kind of issue is why, in my opinion, we should ultimately parse both backend and field — but backend should never be deduced from field. I think, as mentioned elsewhere, that initially having compound backend+field names is fine. In an 'ideal' world, that might become unwieldy when we have enough backends and fields with overlap, that explicitly naming every supported combination is tedious.